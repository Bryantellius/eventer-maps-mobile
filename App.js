import { StatusBar } from "expo-status-bar";
import {
  Image,
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  Platform,
} from "react-native";
import * as ImagePicker from "expo-image-picker";
import * as Sharing from "expo-sharing";
import { useState } from "react";

export default function App() {
  let [selectedImage, setSelectedImage] = useState(null);

  let openImagePickerAsync = async () => {
    let permissionResult = await ImagePicker.requestCameraPermissionsAsync();

    if (permissionResult.granted === false) {
      return alert("Permission to access camera roll is required!");
    }

    let pickerResult = await ImagePicker.launchImageLibraryAsync();

    if (pickerResult.cancelled !== true) {
      setSelectedImage({ uri: pickerResult.uri });
    }
  };

  let openShareDialogAsync = async () => {
    if (Platform.OS === "web") {
      return alert("Uh oh, sharing isn't available on your platform!");
    }

    await Sharing.shareAsync(selectedImage.uri);
  };

  if (selectedImage) {
    return (
      <View style={styles.container}>
        <Text style={styles.heading}>Heading</Text>

        <Image source={{ uri: selectedImage.uri }} style={styles.image} />

        <TouchableOpacity style={styles.button} onPress={openShareDialogAsync}>
          <Text style={styles.buttonText}>Share Photo</Text>
        </TouchableOpacity>
        <StatusBar style="auto" />
      </View>
    );
  } else {
    return (
      <View style={styles.container}>
        <Text style={styles.heading}>Heading</Text>

        <Text style={styles.text}>
          To share a photo from your phone with a friend, just press the button
          below!
        </Text>

        <TouchableOpacity style={styles.button} onPress={openImagePickerAsync}>
          <Text style={styles.buttonText}>Pick a Photo</Text>
        </TouchableOpacity>
        <StatusBar style="auto" />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
    padding: 15,
  },
  heading: {
    color: "orange",
    padding: 5,
    fontSize: 40,
    borderBottomColor: "gray",
    borderBottomWidth: 2,
  },
  text: {
    color: "#222",
    fontSize: 18,
    marginTop: 15,
    marginBottom: 15,
  },
  image: {
    width: 300,
    height: 300,
    resizeMode: "cover",
    borderColor: "black",
    borderWidth: 2,
  },
  button: {
    backgroundColor: "orange",
    padding: 10,
    borderRadius: 8,
    marginVertical: 10,
  },
  buttonText: {
    color: "#FFFFFF",
    fontSize: 20,
  },
});
