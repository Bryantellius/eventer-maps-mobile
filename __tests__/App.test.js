import App from "../App";
import renderer from "react-test-renderer";

describe("<App />", () => {
  it("has 1 child", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children.length).toBe(3);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
