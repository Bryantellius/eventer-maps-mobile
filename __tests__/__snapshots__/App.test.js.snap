// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "padding": 15,
    }
  }
>
  <Text
    style={
      Object {
        "borderBottomColor": "gray",
        "borderBottomWidth": 2,
        "color": "orange",
        "fontSize": 40,
        "padding": 5,
      }
    }
  >
    Heading
  </Text>
  <Text
    style={
      Object {
        "color": "#222",
        "fontSize": 18,
        "marginBottom": 15,
        "marginTop": 15,
      }
    }
  >
    To share a photo from your phone with a friend, just press the button below!
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "orange",
        "borderRadius": 8,
        "marginVertical": 10,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 20,
        }
      }
    >
      Pick a Photo
    </Text>
  </View>
</View>
`;
